Improve optional interfaces description & mains pairing

lategoodbye opened the pull request:
Adapt the hardware chapter to the latest EVerest firmware and improve the description of optional interfaces.

lategoodbye committed:
Hardware: Update mains pairing to latest firmware

Because of security reasons the pairing via MME doesn't work anymore.
So update to the latest workflow.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>

lategoodbye committed:
Hardware: Clarify EIA-485 description

Every EIA-485 interface has a detailed description further below,
so add links to these sections. Also add the word "optional" to
clarify which of these interfaces is optional.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>

lategoodbye commented:
Sorry, I missed the other optional interfaces. Now this should be addressed.

barsnick commented on docs/source/hardware.rst:
```diff
@@ -652,7 +652,7 @@ Following table shows the related configuration:
 ======
 
-This is a generic 1-Wire interface. It is realised with an I2C to 1-wire bridge.
+Some product variants provides a `X4 - 1-Wire`_ interface. It is realised with an I2C to 1-wire bridge.
```
Nit grammar
```suggestion
Some product variants provide a `X4 - 1-Wire`_ interface. It is realised with an I2C to 1-wire bridge.
```

mhei replied:
"provides" -> "provide" and "a" -> "an"

barsnick reviewed (changes requested):
One minor nit.

mhei reviewed (changes requested):
two minor spelling, otherwise LGTM

lategoodbye committed:
Hardware: Clarify optional interfaces X3,X4,X12

Fan output, 1-Wire and X12 are also optional. So clarify the description
here.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>

lategoodbye commented:
Should be fixed now
